### DISCUSSION TITLE
`hostType: "go"` has no effect; private go modules not supported

### DISCUSSION THREAD

**Discussion by buschtoens** [renovatebot/renovate#32562] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I've tried following [App Secrets](https://docs.renovatebot.com/mend-hosted/app-secrets/) to enable updating private Go modules.

I've tried all the `hostRules` I could think of, but none seem to work:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended"
  ],
  "hostRules": [
    {
      "matchHost": "github.com",
      "hostType": "github",
      "token": "{{ secrets.GITHUB_TOKEN }}"
    },
    {
      "matchHost": "api.github.com",
      "hostType": "github",
      "token": "{{ secrets.GITHUB_TOKEN }}"
    },
    {
      "matchHost": "github.com",
      "hostType": "go",
      "token": "{{ secrets.GITHUB_TOKEN }}"
    },
    {
      "matchHost": "github.com",
      "token": "{{ secrets.GITHUB_TOKEN }}"
    }
  ],
  "postUpdateOptions": ["gomodTidy"]
}
```

The `GITHUB_TOKEN` is set up as an organization-level secret and does show up in the `Combined config` in the log. It is a GitHub personal access token with `repo` scope. The user has the permission to read the repository in question.

Yet renovate fails to lookup the dependency:
```
Failed to look up go package github.com/ClarkSource/cib
{
  "dependency": "github.com/ClarkSource/cib"
  "packageFile": "go.mod"
}
```

Does anyone have a working go setup for private modules with the Mend.io-hosted Renovate app?

**Comment by rarkins** [comment]
Most importantly: if you're running on github.com, never ever set tokens like this for github.com or api.github.com. That risks overriding the app's own token! Any such token should be scoped more narrowly. But before that, what reason leads you to need this in the first place?
1. This private go module lives in a different github org to your repo? or
2. It's the same github org, but you have a reason why you can't install the app into the go module repo, or
3. You are referencing a private repo go module from a public github repo?
4. Something else? Please elaborate if so

**Comment by buschtoens** [comment]
@rarkins Thanks for the quick reply!

It would be scenario 2.

Both repositories are in the same organization (@ClarkSource) and private. `ClarkSource/yaari-cli` (the repo I'm trying to configure renovate in) depends on `ClarkSource/cib`.

Initially I tried it without any `hostRules` or secrets, but that did not work (same error as above). It works fine for all other go dependencies which are public.

**Reply by rarkins** [reply, marked as answer]
My recommendation would be to install `ClarkSource/cib` into the app but set it to disabled or silent mode, so that it doesn't generate PRs. That way the existing github.com token the app has will be granted access to the repo by GitHub.

**Reply by buschtoens** [reply]
Ah okay, I think I misunderstood you the first time around. By "app" you are referring to the Renovate GitHub app?

Unfortunately `ClarkSource/cib` is _already_ successfully using Renovate. For `ClarkSource/yaari-cli` it works fine, too, except for the only private dependency on `ClarkSource/cib`.

Edit: Oh, wait a second. I think you're right, I'll verify once more.

**Reply by buschtoens** [reply]
Yes, you were right!

Turns out I was looking at stale Renovate PRs in `cib`, which erroneously lead me to believe it was already/still enabled for `cib`.

After I (re-)added `cib` to the Renovate app, it just works, without any `hostRules` or secrets. ✨ 

Thanks a lot for the support, @rarkins! 🫶
